feat: move hard param to filter config

m-etourneau opened the pull request:
### Description

This PR introduces to move hard coding on ES request to config params on the datahub

jahow committed:
feat(webcomponent): introduce separate standalone search module

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/geonetwork/geonetwork-ui?pullRequest=1093) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/geonetwork/geonetwork-ui?pullRequest=1093) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/geonetwork/geonetwork-ui?pullRequest=1093) it.</sub>

m-etourneau committed:
fix : remove function and add lodash to merge ES state
Adapt the test

m-etourneau committed:
fix : remove function and add lodash to merge ES state
Adapt the test

m-etourneau committed:
add test for merge state filter

m-etourneau committed:
add test for merge state filter

m-etourneau committed:
fix: filter dropdown with configFilter

m-etourneau committed:
fix: filter dropdown with configFilter

fgravin commented:
Hello @m-etourneau , what is the status of the PR ?
If you need me to review something plz ping me because I don't remember if it was ready or not, and I am not notified on changes.

I let you a message on the gitter, thanks !

m-etourneau committed:
fix organisationsCount with the new fonction

m-etourneau committed:
format code

m-etourneau committed:
clean

m-etourneau committed:
clean

m-etourneau committed:
fix test

m-etourneau committed:
fix test

m-etourneau committed:
feat call the new function

m-etourneau committed:
feat call the new function

m-etourneau commented:
Hi @fgravin.
Yes the PR is ready to be review.
But I m not so confident about the getOrganisations(). In particular, in all the part I dont know  the way how to test.
Thank, let me know but I dont receive anything on gitter

fgravin commented:
Hi @m-etourneau could you just rebase your PR please, I am going to test it and check the CI.
Thanks

fgravin commented:
There are also some check issues, usually, you should check the following commands locally
- `npm run format`
- `npm run lint`
- `npm test`

m-etourneau committed:
add filter on search

m-etourneau committed:
add filter on search

m-etourneau commented:
I got some trouble with the lint about circular dependencies because the model SearchFilters is located in the metadata-converter. Should I move it to model/search ? Or not type the SearchFilters ?

fgravin commented:
> I got some trouble with the lint about circular dependencies because the model SearchFilters is located in the metadata-converter. Should I move it to model/search ? Or not type the SearchFilters ?

I will check that, I need to look at the code to answer.
Indeed, shared types should be in shared libraries.

dependabot[bot] committed:
chore(deps): bump axios from 1.7.9 to 1.8.2 in /package

Bumps [axios](https://github.com/axios/axios) from 1.7.9 to 1.8.2.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.7.9...v1.8.2)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

fgravin commented:
Hi, please first fix "npm  run format" command.
For the lint and the circular dep, are you sure that your type is `SearchFilters` and not `FieldFilters` ?

m-etourneau committed:
fix some test cases to use the new input format

m-etourneau committed:
fix some test cases to use the new input format

m-etourneau commented:
I push something, some test case are fixed but others are too tricky for me.
Exemple for this one => https://github.com/geonetwork/geonetwork-ui/blob/1103960a3670d5eb6fd1c4f062104a2ed75b6bd2/libs/feature/search/src/lib/state/effects.spec.ts#L306
I dont understand why the main frame is not set, but the default is set

fgravin commented on libs/common/domain/src/lib/organizations.service.interface.ts:
```diff
@@ -1,12 +1,13 @@
   public abstract getOrganisations(
-    configFilters?: SearchFilters
+    //configFilters?: SearchFilters
+    configFilters?: any
```
We avoid to use `any`, this has a type in the state, it's `FieldFilters`, see

```ts
export interface SearchStateSearch {
  config: {
    aggregations: AggregationsParams
    filters: FieldFilters
    source: FieldName[]
  }
```

fgravin commented on libs/feature/search/src/lib/filter-dropdown/filter-dropdown.component.spec.ts:
```diff
@@ -47,6 +49,14 @@ class FieldsServiceMock {
   getFieldType = jest.fn(() => 'values')
 }
 
+const configFilters: SearchFilters = {
```
Inconsistency in the type, you used `any` above, and it should be `FieldFilters`

fgravin commented on libs/feature/search/src/lib/state/selectors.ts:
```diff
@@ -7,7 +7,11 @@ export const getSearchState =
   getSearchState,
-  (state: SearchState, id: string = DEFAULT_SEARCH_KEY) => state[id]
+  (state: SearchState, id: string = DEFAULT_SEARCH_KEY) => {
+    console.log('Sélecteur getSearchStateSearch - state:', state)
```
Remove all useless `console.log`

fgravin commented on libs/feature/search/src/lib/utils/service/fields.service.spec.ts:
```diff
@@ -43,6 +45,14 @@ class PlatformServiceInterfaceMock {
   })
 }
 
+const configFilters: SearchFilters = {
```
Same, wrong type

fgravin reviewed (commented):
I can have a look at the test but please clean the types and the lint before.
Thanks

m-etourneau committed:
fix searchFilter => FieldFilter and last test

m-etourneau committed:
fix searchFilter => FieldFilter and last test

m-etourneau committed:
clean log

fgravin commented:
Thanks, `format` and `lint` pass.
Still errors in the tests, while shouldn't, eg.

```
 FAIL   api-metadata-converter  libs/api/metadata-converter/src/lib/iso19115-3/iso19115-3.converter.spec.ts (11.057 s)
```

Could you remind the state of your PR ?
Should I look into some specific tests which you could not fix (had effects tests in mind) ? Or everything is ready ?

Thanks

fgravin commented on libs/common/domain/src/lib/repository/records-repository.interface.ts:
```diff
@@ -6,6 +6,7 @@ import {
   SearchParams,
   SearchResults,
 } from '../model/search'
+//import { SearchFilters } from '@geonetwork-ui/api/metadata-converter'
```
remove

fgravin commented on libs/feature/search/src/lib/utils/service/fields.service.ts:
```diff
@@ -101,10 +102,10 @@ export class FieldsService {
   constructor(protected injector: Injector) {}
 
-  getAvailableValues(fieldName: string) {
+  getAvailableValues(fieldName: string, configFilters: SearchFilters) {
```
wrong type

fgravin commented on libs/feature/search/src/lib/utils/service/fields.spec.ts:
```diff
@@ -683,17 +724,22 @@ describe('search fields implementations', () => {
+        )
       })
       it('orders results by descending count', () => {
-        expect(repository.aggregate).toHaveBeenCalledWith({
```
`npm run format` ?

m-etourneau committed:
fix code review comments

m-etourneau committed:
fix code review comments

m-etourneau committed:
fix last code review on mockImplementation

m-etourneau committed:
fix last code review on mockImplementation

m-etourneau commented:
fix for code review done

fgravin commented on libs/feature/search/src/lib/state/effects.spec.ts:
```diff
@@ -49,22 +49,81 @@ import { FavoritesService } from '@geonetwork-ui/api/repository'
+      },
     },
   },
-  main: {
```
Your testes fail because you remove the search state for the `main` search.
Those tests specifically test that the action are forwarded with the input main id (eg. `main`)

m-etourneau committed:
Fix test with main state

m-etourneau committed:
Fix test with main state

AlitaBernachot committed:
feat: allow search of service and reuse

fgravin commented on libs/api/repository/src/lib/gn4/organizations/organizations-from-metadata.service.spec.ts:
```diff
@@ -212,11 +212,15 @@ describe.each(['4.2.2-00', '4.2.3-xx', '4.2.5-xx'])(
         beforeEach(() => {
-          service.organisations$
+          service
+            .getOrganisations({})
```
Add a test with `configFilters`

fgravin commented on libs/api/repository/src/lib/gn4/organizations/organizations-from-metadata.service.ts:
```diff
@@ -152,59 +163,69 @@ export class OrganizationsFromMetadataService
+    configFilters?: FieldFilters
+  ) {
     const semVersion = valid(coerce(gnVersion))
-    return this.esService.getSearchRequestBody({
```
I don't get why github marks that many changes, while there is only one additional line, weird.

fgravin commented on libs/common/domain/src/lib/organizations.service.interface.ts:
```diff
@@ -1,10 +1,12 @@
 import { FieldFilters } from './model/search'
-
 export abstract class OrganizationsServiceInterface {
   public abstract organisations$: Observable<Organization[]>
```
should be removed now that you added `getOrganisations` fcuntion, doesn't it ?

fgravin commented on libs/feature/catalog/src/lib/my-org/my-org.service.spec.ts:
```diff
@@ -86,6 +87,7 @@ describe('MyOrgService', () => {
     })
   })
 
+  //passe pas
```
remove

fgravin commented on libs/feature/search/src/lib/utils/service/fields.service.spec.ts:
```diff
@@ -110,15 +120,18 @@ describe('FieldsService', () => {
+        )
       })
       it('gets the values from the orgs service', () => {
+        console.log('values', values)
```
remove

fgravin commented on libs/feature/search/src/lib/utils/service/fields.spec.ts:
```diff
@@ -480,17 +503,22 @@ describe('search fields implementations', () => {
-              limit: 1000,
-              field: 'tag.default',
-              sort: ['desc', 'count'],
+          expect(repository.aggregate).toHaveBeenCalledWith(
```
weird formatting changes, I wonder how the `npm run format` does not say anything

fgravin reviewed (commented):
A deeper review.
Looks good but still some comments.
I am not happy about how we implemented `getOrganisations` though, looks like not using well the Obersable, I will check again this later.

Thanks

m-etourneau committed:
add configFilter un getRecords / refacto organisation / Add test

m-etourneau committed:
add configFilter un getRecords / refacto organisation / Add test

m-etourneau commented:
I mainly changed the RecordsService from recordsCount$ to function getRecordsCount to add the configFilter.

m-etourneau committed:
Add configFilter in fuzzySearch

m-etourneau committed:
Add configFilter in fuzzySearch

dependabot[bot] committed:
chore(deps): bump axios and @openapitools/openapi-generator-cli

Bumps [axios](https://github.com/axios/axios) and [@openapitools/openapi-generator-cli](https://github.com/OpenAPITools/openapi-generator-cli). These dependencies needed to be updated together.

Updates `axios` from 1.7.9 to 1.8.4
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.7.9...v1.8.4)

Updates `@openapitools/openapi-generator-cli` from 2.15.3 to 2.18.4
- [Release notes](https://github.com/OpenAPITools/openapi-generator-cli/releases)
- [Changelog](https://github.com/OpenAPITools/openapi-generator-cli/blob/master/.releaserc)
- [Commits](https://github.com/OpenAPITools/openapi-generator-cli/compare/v2.15.3...v2.18.4)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
- dependency-name: "@openapitools/openapi-generator-cli"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

jahow committed:
feat: simplify reading the current gn-ui version

jahow committed:
feat(wc): build standalone search script alongside WC one

jahow committed:
feat: make standalone search system work, include docs

cmoinier committed:
feat: new component service capabilities

cmoinier committed:
feat: new component md spatial extent

cmoinier committed:
feat: adaptations to header record

cmoinier committed:
feat: adapt record metadata to service records

cmoinier committed:
feat: adapt md info to service records

cmoinier committed:
feat: pass record kind and return links for service kind

cmoinier committed:
feat: return all service layers in getLayers for service capabilities comp

cmoinier committed:
feat: various UI improvements in record header

cmoinier committed:
feat: route to /service and error if dataset path used

cmoinier committed:
feat: display service records from elasticsearch results

cmoinier committed:
feat: harmonize ServiceEndpoint model to other link types (url and accessServiceProtocol)

cmoinier committed:
feat: use proper typing to handle both DatasetRecords and ServiceRecords

cmoinier committed:
feat: translations

cmoinier committed:
feat: e2e testing

tkohr committed:
fix(favorite-star): make sure tooltip translation is always loaded

also when record page is accessed directly

tkohr committed:
feat(favorite-star): display in secondary color in results list

according to new mockups. make gn-ui-btn-no-outline effects use secondary

tkohr committed:
feat(anchor-link): fine tune IntersectionObserver margins

and add gnUiAnchorLinkInViewClass to storybook

tkohr committed:
refactor(navigation-bar): define shadow via tailwind interpolation instead of config

tkohr committed:
chore(nav-bar): adapt anchors after rebase

cmoinier committed:
feat: make fuzzy search work with services

tkohr committed:
test(e2e): fix tests

cmoinier committed:
fzat: rebase new section org for record page and harmonize title and spacing to service page

cmoinier committed:
feat: do not display geo badge for service records

tkohr committed:
feat(nav-bar): adapt width, font size, text align

jahow committed:
ci: simplify check for affected e2e tasks

Also cache full node_modules folder and skip npm ci, this saves around 30-40s

Also cache cypress binaries and nx cache

cmoinier committed:
feat: route correctly for tab opening

cmoinier committed:
fix: change link section display conditions for services

cmoinier committed:
feat: remove layerInfo if search doesnt include layer + fields translations

cmoinier committed:
fix: fix e2e test

ronitjadhav committed:
add support for reading tms protocol and getting style info

ronitjadhav committed:
Added tms styles in the map-view dropdown

ronitjadhav committed:
Update apiLinks$ to list tms styles

cmoinier committed:
feat: fix ut and display of geodata badge + e2e fix

tkohr committed:
feat(nav-bar): display a fixed title for mobile

as dynamic titles do not update on scroll

tkohr committed:
feat(nav-bar): adapt bg and hover colors

tkohr committed:
fix(nav-bar): add padding to prevent label to move

tkohr committed:
refactor(nav-bar): move css variable to css sheet

cmoinier committed:
feat: make ui component for spatial extent common to dh/editor

cmoinier committed:
feat: make whole favorite star btn clickable

cmoinier committed:
feat: use new spatial extent component and fix border radius and margins

cmoinier committed:
feat: various UI fixes

cmoinier committed:
fix: fix unit tests

tkohr committed:
fix(nav-bar): make nav-bar sticky and move it outside header

tkohr committed:
feat(nav-bar): update mobile menu title when scrolling

and add gnUiAnchorLinkOutOfViewClass to anchor-link.directive

tkohr committed:
fix(e2e): adapt selectors

tkohr committed:
fix(nav-bar): move padding to prevent label to move to gnUiAnchorLinkOutOfViewClass

cmoinier committed:
feat: fixes in service capabilities UI

tkohr committed:
test(e2e): fix tests

jahow committed:
docs: add e2e testing guidelines

jahow committed:
e2e: only do retries on CI, not locally

jahow committed:
e2e(me): rework dashboard test to be faster

tkohr committed:
refactor(gn-ui-button): allow configuring border width

and remove previously added button type 'no-outline'

tkohr committed:
fix(navigation-bar): remove margin

cmoinier committed:
feat: lock the non-dataset records

jahow committed:
e2e(me): reword record-actions to be faster

AlitaBernachot committed:
feat: doc for basemap, max feat, ogcc cache, app banner

cmoinier committed:
feat: adapt nav bar to services and fix buttons

jahow committed:
e2e(me): rework header fields test to be faster

cmoinier committed:
fix: fix e2e tests after editor filtering and fix button style in editor

cmoinier committed:
feat: display reuse records

cmoinier committed:
feat: routing for reuse records

cmoinier committed:
feat: UI adaptations for reuse page and general UI precisions

AlitaBernachot committed:
fix: rework display

tkohr committed:
chore: npm run format

LHBruneton-C2C committed:
fix: properly parse settings_url from global config

jahow committed:
e2e(me): rework about fields test to be faster

Make the image smaller

cmoinier committed:
feat: e2e testing

cmoinier committed:
feat: fixes from review

cmoinier committed:
feat: translations

cmoinier committed:
fix: color of keywords title

cmoinier committed:
feat: keywords badges color and move css clases

cmoinier committed:
feat: rework preview component

cmoinier committed:
feat: place overview in header and border to catalog name

cmoinier committed:
feat: adjust capabilities component background to new height

dependabot[bot] committed:
chore(deps): bump vite and @angular-devkit/build-angular

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) to 5.4.18 and updates ancestor dependencies [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) and [@angular-devkit/build-angular](https://github.com/angular/angular-cli). These dependencies need to be updated together.


Updates `vite` from 5.4.6 to 5.4.18
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.18/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.18/packages/vite)

Updates `vite` from 5.4.11 to 5.4.18
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.18/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.18/packages/vite)

Updates `@angular-devkit/build-angular` from 18.2.12 to 18.2.18
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Changelog](https://github.com/angular/angular-cli/blob/main/CHANGELOG.md)
- [Commits](https://github.com/angular/angular-cli/compare/18.2.12...18.2.18)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 5.4.18
  dependency-type: indirect
- dependency-name: vite
  dependency-version: 5.4.18
  dependency-type: indirect
- dependency-name: "@angular-devkit/build-angular"
  dependency-version: 18.2.18
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

jahow committed:
e2e(me): try to make tests more reliable

jahow committed:
e2e: install cypress-terimanl-report for better e2e output (inc. commands)

jahow committed:
feat(standalone-search): fixes post-review

jahow committed:
e2e(dh): rework dataset detail test to be faster

AlitaBernachot committed:
fix: select all option

AlitaBernachot committed:
doc: add doc for ogcc cache and rephrase some sections

tkohr committed:
feat(gn4.field.mapper): make TMS protocol test more generic

and some fixtures/tests

tkohr committed:
feat(md-view.facade): do not add sorting of mapApiLinks$

AlitaBernachot committed:
fix: icon size

tkohr committed:
feat(external-viewer-button): accept links and styles as input

but do not use styles yet

tkohr committed:
test(map-view.component): add and adapt tests for TMS

tkohr committed:
test(gn4.field.mapper): make TMS protocol case insensitive

and fix fixture

AlitaBernachot committed:
fix: use grid instead of flex to align img and small screen ok

AlitaBernachot committed:
fix: read more text color

dependabot[bot] committed:
chore(deps): bump @nestjs/common from 10.1.3 to 10.4.16

Bumps [@nestjs/common](https://github.com/nestjs/nest/tree/HEAD/packages/common) from 10.1.3 to 10.4.16.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.16/packages/common)

---
updated-dependencies:
- dependency-name: "@nestjs/common"
  dependency-version: 10.4.16
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

AlitaBernachot committed:
fix: remove border on map and minimize spaces

AlitaBernachot committed:
fix: header grid width

AlitaBernachot committed:
fix: prevent blink when empty thumbnail

AlitaBernachot committed:
fix: external link btn alignement when mobile

AlitaBernachot committed:
fix: margin for data integration component

jahow committed:
docs: improve e2e testing guidelines

AlitaBernachot committed:
fix: align titles left

AlitaBernachot committed:
fix: star button error in console

AlitaBernachot committed:
fix: h1, h2, h3 title for accessibility

AlitaBernachot committed:
fix: rollback on map border, mockups are updated

AlitaBernachot committed:
fix: no white text when hovering when no click

AlitaBernachot committed:
fix: unit test

AlitaBernachot committed:
fix: e2e

AlitaBernachot committed:
doc: remove useless link

AlitaBernachot committed:
doc: fix wording

AlitaBernachot committed:
fix: remove unused import and unused mapper

AlitaBernachot committed:
refacto: move new field above legacy and improve jsdoc

fgravin commented on libs/feature/search/src/lib/fuzzy-search/fuzzy-search.component.ts:
```diff
@@ -30,14 +30,23 @@ export class FuzzySearchComponent implements OnInit {
-    this.recordsRepository
-      .fuzzySearch(query)
-      .pipe(map((result) => result.records))
+  autoCompleteAction = (query: string) => {
```
```ts
  autoCompleteAction = (query: string): Observable<CatalogRecord[]> => {
    return this.searchFacade.configFilters$.pipe(
      switchMap((configFilters) =>
        this.recordsRepository.fuzzySearch(query, configFilters)
      ),
      map((result) => result.records)
    )
  }
```

AlitaBernachot committed:
fix: revert useless edits on file

m-etourneau committed:
fixed code review

m-etourneau committed:
fixed code review

fgravin commented on apps/datahub/src/app/home/search/search-filters/search-filters.component.ts:
```diff
@@ -50,6 +51,8 @@ export class SearchFiltersComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
+    this.searchFacade.setConfigFilters(globalConfigFilters)
```
should not be needed here either.

fgravin commented on apps/datahub/src/app/home/search/search-page/search-page.component.ts:
```diff
@@ -22,6 +23,7 @@ export class SearchPageComponent implements OnInit {
   ) {}
 
   ngOnInit() {
+    this.searchFacade.setConfigFilters(globalConfigFilters)
```
here should be the only place to do it.
you can use the chaining 
```ts
this.searchFacade.setResultsLayout('ROW')
     .setConfigFilters(globalConfigFilters)
```

fgravin reviewed (commented):
Try my last comment about the `.setConfigFilters(globalConfigFilters)` thing please.

AlitaBernachot committed:
refacto: create new comp for search kind filter

AlitaBernachot committed:
fix: ut

tkohr committed:
refactor(mdview.facade): simplify and unify TMS logic in facade

also for map-view.component.
fix and move GPFDL sort to record-apis.component.
use concatMap instead of mergeMap to ensure keeping received link order

AlitaBernachot committed:
fix: use the right ts type

AlitaBernachot committed:
refacto: move test function from tpl to ts file

AlitaBernachot committed:
fix: missing export for npm package

tkohr committed:
feat(map-view): handle TMS without style

and complete related tests in facade at the same time

AlitaBernachot committed:
fix: some margins and overlay here and there...

AlitaBernachot committed:
feat: use new internal link card

AlitaBernachot committed:
fix: section titles now are all h2

AlitaBernachot committed:
fix: dropdown placeholder is bold

AlitaBernachot committed:
fix: size and color for catalog and contact section

AlitaBernachot committed:
fix: add missing test

AlitaBernachot committed:
fix: uts css selector

Guillaume-d-o committed:
Change layout for cards depending on content and screen size (#1172)

AlitaBernachot committed:
fix: e2e wallo service is down

AlitaBernachot committed:
fix: after rebase

AlitaBernachot committed:
fix(internal-card): add to fav icon is not aligned with badges

cmoinier committed:
feat: add code into featureCatalog response

cmoinier committed:
feat: parse properties labels in data view

cmoinier committed:
feat: parse properties in map details

cmoinier committed:
feat: hide tab when preview/capabilities not present

cmoinier committed:
fix: unit tests

AlitaBernachot committed:
feat: now use internal card as default record row preview

AlitaBernachot committed:
feat: use badge kind everywhere

AlitaBernachot committed:
feat: use grid display in record page

AlitaBernachot committed:
fix(internal-card): allow fluid width

cmoinier committed:
feat: new multilingual panel

cmoinier committed:
feat: open multilingual panel

cmoinier committed:
feat: e2e tests and translations

cmoinier committed:
fix: add translatemodule import in ut

cmoinier committed:
feat: changes from review

AlitaBernachot committed:
feat: is geodata in search list

AlitaBernachot committed:
refactor: geodata badge

AlitaBernachot committed:
refactor: use geodata badge everywhere

AlitaBernachot committed:
fix: get online resources in es for related resources

AlitaBernachot committed:
refactor: clean unused isGeodata

AlitaBernachot committed:
fix: size of geo icon when no label

AlitaBernachot committed:
fix: make 'L' layout fluid

AlitaBernachot committed:
refactor: move internal link card insidre related record

AlitaBernachot committed:
fix: remove unused linkHref

Guillaume-d-o committed:
New truncate text component (#1203)

* new truncate text component for cell with large content

AlitaBernachot committed:
feat: open relataed record with the router

AlitaBernachot committed:
feat: add fav to related

AlitaBernachot committed:
fix: wording, there is not only datasets now

cmoinier committed:
feat: refactor display conditions for more clarity

AlitaBernachot committed:
fix: uts

AlitaBernachot committed:
fix: some cleaning and rollbacks

cmoinier committed:
feat: add documentation for section display

cmoinier committed:
feat: add e2e testing for nav bar buttons

AlitaBernachot committed:
fix: ci

cmoinier committed:
fix: make new e2e tests work

dependabot[bot] committed:
chore(deps): bump http-proxy-middleware and @angular-devkit/build-angular

Bumps [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware) to 3.0.5 and updates ancestor dependencies [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware) and [@angular-devkit/build-angular](https://github.com/angular/angular-cli). These dependencies need to be updated together.


Updates `http-proxy-middleware` from 3.0.3 to 3.0.5
- [Release notes](https://github.com/chimurai/http-proxy-middleware/releases)
- [Changelog](https://github.com/chimurai/http-proxy-middleware/blob/master/CHANGELOG.md)
- [Commits](https://github.com/chimurai/http-proxy-middleware/compare/v3.0.3...v3.0.5)

Updates `http-proxy-middleware` from 2.0.7 to 3.0.5
- [Release notes](https://github.com/chimurai/http-proxy-middleware/releases)
- [Changelog](https://github.com/chimurai/http-proxy-middleware/blob/master/CHANGELOG.md)
- [Commits](https://github.com/chimurai/http-proxy-middleware/compare/v3.0.3...v3.0.5)

Updates `@angular-devkit/build-angular` from 18.2.18 to 18.2.19
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Changelog](https://github.com/angular/angular-cli/blob/main/CHANGELOG.md)
- [Commits](https://github.com/angular/angular-cli/compare/18.2.18...18.2.19)

---
updated-dependencies:
- dependency-name: http-proxy-middleware
  dependency-version: 3.0.5
  dependency-type: indirect
- dependency-name: http-proxy-middleware
  dependency-version: 3.0.5
  dependency-type: indirect
- dependency-name: "@angular-devkit/build-angular"
  dependency-version: 18.2.19
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

ronitjadhav committed:
Update handling of TMS without styles according to geospatial-sdk

ronitjadhav committed:
Prevent infinite loop in TMS links processing

cmoinier committed:
fix: fix table and chart behavior

cmoinier committed:
fix: fix chart view behavior further and fix unit tests

ronitjadhav committed:
Improve visibility of spatial extent bounding boxes

Use a semi-transparent fill color to make spatial extents more
distinguishable, especially for large areas like global datasets.

ronitjadhav committed:
Update the unit test for spatial-extent

tkohr committed:
fix(nav-bar): set height explicitly

to fix nav bar flickering/position on firefox being stricter with sticky

tkohr committed:
refactor(nav-bar): move css to nav-bar component

AlitaBernachot committed:
fix: clean unused import

AlitaBernachot committed:
fix: add a link instead of just a click

AlitaBernachot committed:
fix: margin related section title and lighter separator color

AlitaBernachot committed:
fix: remove unused component

AlitaBernachot committed:
fix: update storybook cas no geodata badge

cmoinier committed:
fix chart and refacto

AlitaBernachot committed:
fix: add tooltip on badge when no label

AlitaBernachot committed:
fix: add tooltip on badge when no label

AlitaBernachot committed:
fix: move record kind marker in kind badge

AlitaBernachot committed:
fix: broken wc, add optional inject token

AlitaBernachot committed:
fix: e2e

cmoinier committed:
feat: make chart label prettier too

AlitaBernachot committed:
fix: add missing block list

AlitaBernachot committed:
fix: ut

AlitaBernachot committed:
fix: e2e

AlitaBernachot committed:
fix: block list size

AlitaBernachot committed:
fix: e2e

tkohr committed:
fix(read-parts): add MimeFileType tag to CharacterString fallback

tkohr committed:
fix(gn4-repo): use record xml when saving record

AlitaBernachot committed:
fix: ut

dependabot[bot] committed:
chore(deps): bump vite from 5.4.18 to 5.4.19

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.4.18 to 5.4.19.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.19/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.19/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 5.4.19
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Guillaume-d-o committed:
Display and filter feature catalogs (#1206)

* Display and filter feature catalogs

ronitjadhav committed:
WIP

cmoinier committed:
feat: fixes on service capabilities

AlitaBernachot committed:
fix: rollback, keep 3 related records

jahow committed:
test: use ES2016 as target for unit tests

Angular requires ES2016 as a target for the async/await syntax to be patched.

See https://github.com/angular/angular/issues/31730

AlitaBernachot committed:
fix: border on thumbnail img internal card

ronitjadhav committed:
Fix unit tests

ronitjadhav committed:
cleanup - remove unnecessary comments and logs

AlitaBernachot committed:
fix: uts after rollback 3 records

cmoinier committed:
feat: center clear icon and add translations

tkohr committed:
test(read-parts): use MimeFileType in one fixture

cmoinier committed:
feat: update italian translations

AlitaBernachot committed:
fix: after rebase

tkohr committed:
fix(backport workflow): convert array to string and inverse logic for label check

LHBruneton-C2C committed:
feat(fuzzy-search): force start align on suggestions to prevent herited center align

m-etourneau committed:
global filter on the top call stack

m-etourneau committed:
global filter on the top call stack

m-etourneau committed:
rebase global configFilter

m-etourneau committed:
rebase

m-etourneau committed:
fix test
